pkg/controller/volume/cache/actual_state_of_world_test.go

```diff
@@ -649,34 +757,135 @@ func Test_MarkDesireToDetach_Positive_MarkedVerifySafeToDetachPreserved(t *testi
 		t.Fatalf("len(attachedVolumes) Expected: <1> Actual: <%v>", len(attachedVolumes))
 	}
 
-	verifyAttachedVolume(t, attachedVolumes, generatedVolumeName, volumeName, nodeName, true /* expectedSafeToDetach */, true /* expectNonZeroDetachRequestedTime */)
+	verifyAttachedVolume(t, attachedVolumes, generatedVolumeName, string(volumeName), nodeName, false /* expectedMountedByNode */, true /* expectNonZeroDetachRequestedTime */)
 }
 
 func verifyAttachedVolume(
 	t *testing.T,
 	attachedVolumes []AttachedVolume,
-	expectedVolumeName,
-	expectedVolumeSpecName,
+	expectedVolumeName api.UniqueDeviceName,
+	expectedVolumeSpecName string,
 	expectedNodeName string,
-	expectedSafeToDetach,
+	expectedMountedByNode,
 	expectNonZeroDetachRequestedTime bool) {
 	for _, attachedVolume := range attachedVolumes {
 		if attachedVolume.VolumeName == expectedVolumeName &&
 			attachedVolume.VolumeSpec.Name() == expectedVolumeSpecName &&
 			attachedVolume.NodeName == expectedNodeName &&
-			attachedVolume.SafeToDetach == expectedSafeToDetach &&
+			attachedVolume.MountedByNode == expectedMountedByNode &&
 			attachedVolume.DetachRequestedTime.IsZero() == !expectNonZeroDetachRequestedTime {
 			return
 		}
 	}
 
 	t.Fatalf(
-		"attachedVolumes (%v) should contain the volume/node combo %q/%q with SafeToDetach=%v and NonZeroDetachRequestedTime=%v. It does not.",
+		"attachedVolumes (%v) should contain the volume/node combo %q/%q with MountedByNode=%v and NonZeroDetachRequestedTime=%v. It does not.",
 		attachedVolumes,
 		expectedVolumeName,
 		expectedNodeName,
-		expectedSafeToDetach,
+		expectedMountedByNode,
 		expectNonZeroDetachRequestedTime)
 }
 
-// t.Logf("attachedVolumes: %v", asw.GetAttachedVolumes()) // TEMP
+func Test_GetAttachedVolumesForNode_Positive_NoVolumesOrNodes(t *testing.T) {
+	// Arrange
+	volumePluginMgr, _ := controllervolumetesting.GetTestVolumePluginMgr((t))
+	asw := NewActualStateOfWorld(volumePluginMgr)
+	node := "random"
+
+	// Act
+	attachedVolumes := asw.GetAttachedVolumesForNode(node)
+
+	// Assert
+	if len(attachedVolumes) != 0 {
+		t.Fatalf("len(attachedVolumes) Expected: <0> Actual: <%v>", len(attachedVolumes))
+	}
+}
+
+func Test_GetAttachedVolumesForNode_Positive_OneVolumeOneNode(t *testing.T) {
+	// Arrange
+	volumePluginMgr, _ := controllervolumetesting.GetTestVolumePluginMgr((t))
+	asw := NewActualStateOfWorld(volumePluginMgr)
+	volumeName := api.UniqueDeviceName("volume-name")
+	volumeSpec := controllervolumetesting.GetTestVolumeSpec(string(volumeName), volumeName)
+	nodeName := "node-name"
+	generatedVolumeName, addErr := asw.AddVolumeNode(volumeSpec, nodeName)
+	if addErr != nil {
+		t.Fatalf("AddVolumeNode failed. Expected: <no error> Actual: <%v>", addErr)
+	}
+
+	// Act
+	attachedVolumes := asw.GetAttachedVolumesForNode(nodeName)
+
+	// Assert
+	if len(attachedVolumes) != 1 {
```

Comment:
for stuff like this i like to use this syntax:

```go
if l := len(attachedVolumes); l != 1 {
  t.Fatalf("bla bla %v", l)
}
```

..but you can do that in a follow up if you so choose.